nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/main/java/org/apache/nifi/processors/iceberg/converter/GenericDataConverters.java

```diff
@@ -47,45 +51,106 @@
+            }
         }
     }
 
-    static class TimeConverter extends DataConverter<Time, LocalTime> {
+    static class TimeConverter extends DataConverter<Object, LocalTime> {
+
+        private final String timeFormat;
+
+        public TimeConverter(final String format) {
+            this.timeFormat = format;
+        }
 
         @Override
-        public LocalTime convert(Time data) {
-            return data.toLocalTime();
+        public LocalTime convert(Object data) {
+            if (data instanceof Time) {
+                return ((Time) data).toLocalTime();
+            }
+            return DataTypeUtils.toTime(data, () -> StringUtils.isEmpty(timeFormat)
+                    ? null
+                    : DataTypeUtils.getDateFormat(timeFormat), null).toLocalTime();
         }
     }
 
-    static class TimestampConverter extends DataConverter<Timestamp, LocalDateTime> {
+    static class TimestampConverter extends DataConverter<Object, LocalDateTime> {
+
+        private final DataType dataType;
+
+        public TimestampConverter(final DataType dataType) {
+            this.dataType = dataType;
+        }
 
         @Override
-        public LocalDateTime convert(Timestamp data) {
-            return data.toLocalDateTime();
+        public LocalDateTime convert(Object data) {
+            final Timestamp convertedTimestamp = (Timestamp) DataTypeUtils.convertType(data, RecordFieldType.TIMESTAMP.getDataType(dataType.getFormat()),
+                    null, null, () -> DataTypeUtils.getDateFormat(dataType.getFormat()), null);
+            return convertedTimestamp.toLocalDateTime();
         }
     }
 
-    static class TimestampWithTimezoneConverter extends DataConverter<Timestamp, OffsetDateTime> {
+    static class TimestampWithTimezoneConverter extends DataConverter<Object, OffsetDateTime> {
+
+        private final DataType dataType;
+
+        public TimestampWithTimezoneConverter(final DataType dataType) {
+            this.dataType = dataType;
+        }
 
         @Override
-        public OffsetDateTime convert(Timestamp data) {
-            return OffsetDateTime.ofInstant(data.toInstant(), ZoneId.of("UTC"));
+        public OffsetDateTime convert(Object data) {
+            final Timestamp convertedTimestamp = (Timestamp) DataTypeUtils.convertType(data, RecordFieldType.TIMESTAMP.getDataType(dataType.getFormat()),
+                    null, null, () -> DataTypeUtils.getDateFormat(dataType.getFormat()), null);
+            return OffsetDateTime.ofInstant(convertedTimestamp.toInstant(), ZoneId.of("UTC"));
         }
     }
 
-    static class UUIDtoByteArrayConverter extends DataConverter<UUID, byte[]> {
+    static class UUIDtoByteArrayConverter extends DataConverter<Object, byte[]> {
 
         @Override
-        public byte[] convert(UUID data) {
+        public byte[] convert(Object data) {
+            final UUID uuid;
```

Comment:
The` DataTypeUtils.toUUID` method contains these UUID related logic so you can replace this part with that method call.